TSQL_Parser/TSQL_Parser/Expressions/Parsers/TSQLValuesExpressionParser.cs

```diff
@@ -0,0 +1,53 @@
+﻿using System;
+
+using TSQL.Clauses.Parsers;
+using TSQL.Tokens;
+
+namespace TSQL.Expressions.Parsers
+{
+	internal class TSQLValuesExpressionParser
+	{
+		public TSQLValuesExpression Parse(ITSQLTokenizer tokenizer)
+		{
+			TSQLValuesExpression valuesExpression = new TSQLValuesExpression();
+
+			if (!tokenizer.Current.IsKeyword(TSQLKeywords.VALUES))
+			{
+				throw new InvalidOperationException("VALUES expected.");
+			}
+
+			valuesExpression.Tokens.Add(tokenizer.Current);
+
+			int nestedLevel = 0;
+
+			while (
+				tokenizer.MoveNext() &&
+				!tokenizer.Current.IsCharacter(TSQLCharacters.Semicolon) &&
+				!(
+					nestedLevel == 0 &&
+					tokenizer.Current.IsCharacter(TSQLCharacters.CloseParentheses)
+				) &&
+				(
+					nestedLevel > 0 ||
+					tokenizer.Current.Type != TSQLTokenType.Keyword ||
+					(
+						tokenizer.Current.Type == TSQLTokenType.Keyword &&
+						!tokenizer.Current.AsKeyword.Keyword.In
```

Comment:
I'd like to avoid stop words when possible, especially for an expression that can be found in many different situations in the language.

Couldn't we just end on a CloseParenthesis when not followed by a comma? [Table Value Constructor](https://docs.microsoft.com/en-us/sql/t-sql/queries/table-value-constructor-transact-sql?view=sql-server-ver15)